redundant-bmc/src/active_role_handler.cpp

```diff
@@ -36,6 +37,67 @@ sdbusplus::async::task<> ActiveRoleHandler::start()
 
     // TODO: Create an error if no redundancy
 
+    startSiblingWatches();
+
+    co_return;
+}
+
+void ActiveRoleHandler::siblingStateChange(BMCState state)
+{
+    if (state == BMCState::Quiesced)
+    {
+        lg2::error("Sibling BMC went to Quiesce, disabling redundancy");
+        redMgr.determineAndSetRedundancy();
+    }
+}
+
+void ActiveRoleHandler::siblingHBChange(bool hb)
+{
+    if (hb)
+    {
+        siblingHBTimer.stop();
+        ctx.spawn(siblingHBStarted());
+    }
+    else
+    {
+        lg2::info("Sibling BMC heartbeat lost");
+        if (redundancyInterface.redundancy_enabled())
+        {
+            lg2::info(
+                "Disabling redundancy in {TIME} minutes if sibling heartbeat doesn't recover",
+                "TIME", siblingHBTimeout.count());
+            siblingHBTimer.start(siblingHBTimeout);
+        }
+    }
+}
+
+void ActiveRoleHandler::siblingHBCritical()
+{
+    lg2::error("Sibling heartbeat timer expired, disabling redundancy");
+    redMgr.determineAndSetRedundancy();
+}
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> ActiveRoleHandler::siblingHBStarted()
+{
+    lg2::info("Passive BMC heartbeat started");
+
+    stopSiblingWatches();
+
+    co_await sdbusplus::async::execution::when_all(
+        sibling.waitForSiblingRole(), sibling.waitForBMCSteadyState());
+
+    // Try to re-enable redundancy if it isn't
+    if (!redundancyInterface.redundancy_enabled())
```

Comment:
I think I need to call it even if it was enabled, because it won't necessarily be able to be enabled again.